fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/DbSearchIndexImpl.java

```diff
@@ -252,7 +378,51 @@ public void removeFromIndex(final FedoraId fedoraId) {
 
     @Override
     public void reset() {
-        jdbcTemplate.update("TRUNCATE TABLE " + SIMPLE_SEARCH_TABLE, Collections.EMPTY_MAP);
+        final var builder = new StringBuilder();
+
+        try {
+            final var conn = this.dataSource.getConnection();
+            final var statement = conn.createStatement();
+            for (var sql : toggleForeignKeyChecks(false)) {
+                statement.addBatch(sql);
+            }
+            statement.addBatch(truncateTable(SEARCH_RESOURCE_RDF_TYPE_TABLE));
+            statement.addBatch(truncateTable(SIMPLE_SEARCH_TABLE));
+            statement.addBatch(truncateTable(SEARCH_RDF_TYPE_TABLE));
+            for (var sql : toggleForeignKeyChecks(true)) {
+                statement.addBatch(sql);
+            }
+            statement.executeBatch();
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+    }
+
+    private List<String> toggleForeignKeyChecks(final boolean enable) {
+
+        if (isPostgres()) {
+            final var statements = new ArrayList<String>();
```

Comment:
Minor note, only change if you're inclined, but this could be rewritten as:

```java
return List.of(
    togglePostgresTriggers(SEARCH_RESOURCE_RDF_TYPE_TABLE, enable),
    togglePostgresTriggers(SEARCH_RDF_TYPE_TABLE, enable),
    togglePostgresTriggers(SIMPLE_SEARCH_TABLE, enable),
);
```